### ISSUE TITLE
Bug: website playground gives TS error when defining variable named `console`

### ISSUE THREAD

**Issue by abrahamguo** [typescript-eslint/typescript-eslint#9646]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=MYewdgzgLgBKkgDYFMYF4YEYDcQ&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJgIwHsOATWgFtE%2BABbc%2B6KImjRu0SODABfEMqA&tsconfig=&tokens=false

### Repro Code

```TypeScript
const console = 1;
```

### tsconfig

```jsonc
{
	"compilerOptions": {
		"module": "nodenext",
		"strict": true
	}
}
```

### Expected Result

No TypeScript errors should be reported in the playground.

When running this same code snippet [locally](https://github.com/abrahamguo/repro/tree/console), no TypeScript errors are reported:

- `npm i`
- `npx tsc`

### Actual Result

A TypeScript error was reported in the playground:
```
2451: Cannot redeclare block-scoped variable 'console'. 1:7 - 1:14
```

### Versions

Live version of website
Typescript 5.5.2 on website/5.5.4 on local

**Comment by bradzacher** [comment]
This is working as expected!
By default the file is seen as a "script" - meaning everything is put the global scope. You can redefine globals cos they're existing variables.

If you convert the code to a module by adding `export {};` then you change the rules and now there's a separate, module scope beneath the child scope within which the variables are placed. Thus you're allowed to redefine globals.
